pkg/base/replicated.go

```diff
@@ -152,7 +157,64 @@ func renderReplicated(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (
 		}
 	}
 
-	return &commonBase, helmBases, nil
+	return &commonBase, helmBases, renderedKotsKinds, nil
+}
+
+func renderKotsKinds(upstreamFiles []upstreamtypes.UpstreamFile, renderedConfig *kotsv1beta1.Config, renderOptions *RenderOptions, builder *template.Builder) (map[string][]byte, error) {
+	fileContentsMap := make(map[string][]byte)
+
+	for _, upstreamFile := range upstreamFiles {
+		for _, doc := range util.ConvertToSingleDocs(upstreamFile.Content) {
+			gvk := OverlySimpleGVK{}
+			if err := yaml.Unmarshal(doc, &gvk); err != nil {
+				continue
+			}
+			gvkString := fmt.Sprintf("%s, Kind=%s", gvk.APIVersion, gvk.Kind)
+
+			if !iskotsAPIVersionKind(gvk) {
+				continue
+			}
+
+			switch gvkString {
+			case "kots.io/v1beta1, Kind=Config":
+				// Use the rendered Config instead of the upstream.
+				kotsKinds := kotsutil.KotsKinds{Config: renderedConfig}
+				configBytes, err := kotsKinds.Marshal("kots.io", "v1beta1", "Config")
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to marshal rendered config")
+				}
+				fileContentsMap[upstreamFile.Path] = []byte(configBytes)
```

Comment:
since `configvalues.yaml` is hardcoded, i don't see why this can't just be:
```suggestion
				fileContentsMap["config.yaml"] = []byte(configBytes)
```

Reply:
actually. never mind.